imgui-glium-renderer/src/lib.rs

```diff
@@ -111,15 +114,13 @@ impl Renderer {
             [0.0, 0.0, -1.0, 0.0],
             [-1.0, 1.0, 0.0, 1.0],
         ];
-        let font_texture_id = self.device_objects.texture.get_id() as usize;
 
         let mut idx_start = 0;
         for cmd in draw_list.cmd_buffer {
-            // We don't support custom textures...yet!
-            assert!(cmd.texture_id as usize == font_texture_id);
-
             let idx_end = idx_start + cmd.elem_count as usize;
 
+            let texture = <Texture as FromImTexture>::from_id(cmd.texture_id);
```

Comment:
This is very unsafe, because we're assuming the texture_id always corresponds to a glium texture, but there's nothing in the main imgui crate functions that guarantees it. This is the main difficulty in custom textures: they are used in renderers but added/declared/registered somewhere else.

Reply:
Yes. It is also unsafe if we move the textures within `imgui`. The the textures are moved, we will get a dangling pointer.

To ensure that texture_id always corresponds to the same type of texture, an idea of mine was to use generics on the ImGui instance as below. But that would break all existing code. That's why I decided to go with a type `AnyTexture` encapsulating a trait object.

```rust
/// `T` is the texture type expected by the backend.
pub struct ImGui<T> {
    ini_filename: Option<ImString>,
    log_filename: Option<ImString>,
    textures: TextureCache<T>,
}
```